### PR TITLE
operator: Ensuring amd64 binaries are built; Correcting distroless image name

### PR DESCRIPTION
**Pull request by Red-GV**
**What this PR does / why we need it**:

This PR ensures that an amd64 binary is built. On M1 Macs, the operator image cannot be utilized because the image crashes with a `exec format error` error. This also adjusts the name of the distroless image used to what is depicted in the table here: https://github.com/GoogleContainerTools/distroless#what-images-are-available.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the `CONTRIBUTING.md` guide
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: operator/Dockerfile [hunk 1]
```diff
@@ -16,11 +16,11 @@ COPY controllers/ controllers/
 COPY internal/ internal/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -mod=readonly -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=readonly -a -o manager main.go
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+FROM gcr.io/distroless/static-debian11:nonroot
```
Comment (periklis) [inline operator/Dockerfile:23]: Why is this change needed? The `-debian11` images are only for debugging. However we use this dockerfile to build production images on [docker.io/grafana/loki-operator](https://hub.docker.com/r/grafana/loki-operator)

File: operator/Dockerfile [hunk 2]
```diff
@@ -16,11 +16,11 @@ COPY controllers/ controllers/
 COPY internal/ internal/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -mod=readonly -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -mod=readonly -a -o manager main.go
```
Comment (periklis) [inline operator/Dockerfile:19]: We use this Dockerfile to build multi-arch production images on [docker.io/grafana/loki-operator](https://hub.docker.com/r/grafana/loki-operator). So we cannot simply pin to `amd64`

**Review by periklis — CHANGES_REQUESTED** [review]

File: operator/Dockerfile (hunk 1 above)
Comment (Red-GV) [reply operator/Dockerfile:23]: Gotcha. I think then I just misunderstood the context behind the names described in the GitHub.

File: operator/Dockerfile (hunk 2 above)
Comment (Red-GV) [reply operator/Dockerfile:19]: Gotcha. Hmm, maybe there should be a note in the developer notes for this then. As is, the `Dockerfile` does not work on an M1 Mac.
